### PR TITLE
[P2P] Remove topic from messaging - Issue #267 - NITS

### PR DESCRIPTION
**Pull request by deblasis**
## Description

Addressing https://github.com/pokt-network/pocket/pull/353#pullrequestreview-1190521434

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## Testing

- [x] `make develop_test`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my changes using the available tooling
- [x] I have updated the corresponding CHANGELOG

### If Applicable Checklist
- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: shared/messaging/envelope_test.go [hunk 1]
```diff
@@ -7,15 +7,15 @@ import (
...
 	someMsg := &DebugMessage{Action: DebugMessageAction_DEBUG_CLEAR_STATE}
 	packedMsg, err := PackMessage(someMsg)
 	require.NoError(t, err)
 
-	unpackedMsg, err := UnpackMessage(packedMsg)
+	unpackedMsg, err := UnpackMessage[*DebugMessage](packedMsg)
 	require.NoError(t, err)
 
-	if !proto.Equal(someMsg, unpackedMsg.(proto.Message)) {
+	if !proto.Equal(someMsg, unpackedMsg) {
```
Comment (Olshansk) [inline shared/messaging/envelope_test.go:18]: `require.True(t, proto.Equal(someMsg, unpackedMsg))`?

Ditto elsewhere

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: shared/messaging/envelope_test.go (hunk 1 above)
Comment (deblasis) [reply shared/messaging/envelope_test.go]: Done, but I can't find the same issue anywhere else 🦮

Comment (Olshansk) [reply shared/messaging/envelope_test.go]: For some reason I thought I saw it in another place. Please ignore and 🦮 on

File: shared/messaging/envelope.go [hunk 2]
```diff
@@ -16,11 +16,11 @@ func PackMessage(message proto.Message) (*PocketEnvelope, error) {
 }
 
 // UnpackMessage extracts the message inside the PocketEnvelope decorating it with typing information
-func UnpackMessage(envelope *PocketEnvelope) (protoreflect.ProtoMessage, error) {
+func UnpackMessage[T protoreflect.ProtoMessage](envelope *PocketEnvelope) (T, error) {
```
Comment (Olshansk) [inline shared/messaging/envelope.go:19]: Thoughts on this change? It'll avoid accidentally returning a non-pointer, which I believe is always what we need if you're calling `UnpackMessage`.

```suggestion
func UnpackMessage[T protoreflect.ProtoMessage](envelope *PocketEnvelope) (*T, error) {
```

Comment (Olshansk) [reply shared/messaging/envelope.go:19]: Sgtm

**Review by Olshansk — APPROVED** [review]

**Pull request merged**